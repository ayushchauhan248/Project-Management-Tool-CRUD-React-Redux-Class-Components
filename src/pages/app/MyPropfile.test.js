import React from "react";
import Myprofile from "./MyProfile";
import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
configure({ adapter: new Adapter() });

describe("My Profile tests", () => {
  it("snapshot testing", () => {
    const component = shallow(<Myprofile />); // it return shallow object
    console.log(component.debug()); // debug will provide us full HTML structure
    expect(component).toMatchSnapshot();
  });

  it("should render name correct", () => {
    const wrapper = shallow(<Myprofile />);
    const span = wrapper.find("#msg-btn");
    const res = span.text();
    expect(res).toBe("Message");
  });

  it("should render designation correct", () => {
    const wrapper = shallow(<Myprofile />);
    const span = wrapper.find("#follow-btn");
    const res = span.text();
    expect(res).toEqual("Follows");
  });
});
