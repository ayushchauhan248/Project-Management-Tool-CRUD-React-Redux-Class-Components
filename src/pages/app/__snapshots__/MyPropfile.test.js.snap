// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Profile tests snapshot testing 1`] = `
<div
  id="bd"
>
  <div
    id="contain"
  >
    <div
      id="cover-photo"
    >
      <img
        alt="profileImage"
        id="profile"
        src="https://upload.wikimedia.org/wikipedia/commons/9/99/Sample_User_Icon.png"
      />
    </div>
    <span
      id="profile-name"
    >
      Ayush Chauhan
    </span>
    <p
      id="about"
    >
      Front-end developer And Back-end developer
    </p>
    <button
      class="buttn"
      id="msg-btn"
    >
      Message
    </button>
    <button
      id="follow-btn"
    >
      Follows
    </button>
    <div>
      <i
        class="fab fa-facebook-f"
      />
      <i
        class="fab fa-instagram"
      />
      <i
        class="fab fa-youtube"
      />
      <i
        class="fab fa-twitter"
      />
    </div>
  </div>
</div>
`;
